tests/sqlAst/data/sql-ast-narrowing.php

```diff
@@ -7,6 +7,61 @@
 
 class Foo
 {
+    public function whereIsNotNull(PDO $pdo): void
+    {
+        $stmt = $pdo->query('SELECT c_json FROM typemix');
+        assertType('PDOStatement<array{c_json: string|null, 0: string|null}>', $stmt);
+
+        $stmt = $pdo->query('SELECT c_json FROM typemix WHERE c_json IS NOT NULL');
+        assertType('PDOStatement<array{c_json: string, 0: string}>', $stmt);
+
+        // selected with an alias
+        $stmt = $pdo->query('SELECT c_json as col FROM typemix WHERE c_json IS NOT NULL');
+        assertType('PDOStatement<array{col: string, 0: string}>', $stmt);
+
+        // affects input to a function
+        $stmt = $pdo->query('SELECT ifnull("1", c_json) as col FROM typemix WHERE c_json IS NOT NULL');
+        assertType('PDOStatement<array{col: string, 0: string}>', $stmt);
+
+        // selected with an asterisk
+        $stmt = $pdo->query('SELECT * FROM typemix WHERE c_json IS NOT NULL');
```

Comment:
another case which can be imlpemented in a separate PR:
`'SELECT nullable-col FROM typemix WHERE non-nullable-col = nullable-col'`
assert select of nullable-col cannot contain null